bin/install_upgrade.escript

```diff
@@ -217,7 +219,12 @@ find_and_link_release_package(Version, RelName) ->
     %% we've found where the actual release package is located
     ReleaseLink = filename:join(["releases", Version,
                                  RelNameStr ++ ".tar.gz"]),
-    TarBalls = filename:join(["releases", RelNameStr ++ "-*" ++ Version ++ "*.tar.gz"]),
+    ReleaseNamePattern =
+        case IsEnterprise of
+            false -> RelNameStr;
+            true -> RelNameStr ++ "-enterprise"
+        end,
+    TarBalls = filename:join(["releases", ReleaseNamePattern ++ "-" ++ Version ++ "*.tar.gz"]),
```

Comment:
nit
```suggestion
    TarBalls = filename:join(["releases", ReleaseNamePattern, "-", Version, "*.tar.gz"]),
```

Reply:
I'm afraid that won't have the desired output.  Using `filename:join` will introduce `/` between the parts of the pattern rather than concatenate then.

Reply:
I'll refactor it to use `lists:flatten/1`.

Reply:
:heavy_check_mark: 
https://github.com/emqx/emqx/pull/9612/commits/660d028ebb04c240324d00742cb6249292a9520f